src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Vector128.cs

```diff
@@ -2976,31 +2976,31 @@ public static Vector128<sbyte> Shuffle(Vector128<sbyte> vector, Vector128<sbyte>
         /// <param name="vector">The input vector from which values are selected.</param>
         /// <param name="indices">The per-element indices used to select a value from <paramref name="vector" />.</param>
         /// <returns>A new vector containing the values from <paramref name="vector" /> selected by the given <paramref name="indices" />.</returns>
-        /// <remarks>Unlike Shuffle, this method delegates to the underlying hardware intrinsic without ensuring that <paramref name="indices" /> are normalized to [0, 15].
-        /// On hardware with <see cref="Ssse3" /> support, indices are treated as modulo 16, and if the high bit is set, the result will be set to 0 for that element.
-        /// On hardware with <see cref="AdvSimd.Arm64" /> or <see cref="PackedSimd" /> support, this method behaves the same as Shuffle.</remarks>
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        /// <remarks>Unlike Shuffle, this method delegates to the underlying hardware intrinsic without ensuring that <paramref name="indices"/> are normalized to [0, 15].</remarks>
+        [Intrinsic]
         [CompExactlyDependsOn(typeof(Ssse3))]
         [CompExactlyDependsOn(typeof(AdvSimd))]
         [CompExactlyDependsOn(typeof(AdvSimd.Arm64))]
         [CompExactlyDependsOn(typeof(PackedSimd))]
-        internal static Vector128<byte> ShuffleUnsafe(Vector128<byte> vector, Vector128<byte> indices)
+        public static Vector128<byte> ShuffleUnsafe(Vector128<byte> vector, Vector128<byte> indices)
         {
-            if (Ssse3.IsSupported)
-            {
-                return Ssse3.Shuffle(vector, indices);
-            }
-
-            if (AdvSimd.Arm64.IsSupported)
-            {
-                return AdvSimd.Arm64.VectorTableLookup(vector, indices);
-            }
-
-            if (PackedSimd.IsSupported)
-            {
-                return PackedSimd.Swizzle(vector, indices);
-            }
+            return Shuffle(vector, indices);
```

Comment:
@tannergooding when you get to this PR, can you please confirm when you get to this whether this is the correct approach for falling back to `Shuffle` in case of unoptimised `Shuffle`/`ShuffleUnsafe`? Specifically, will this allow `ShuffleUnsafe` to work correctly (same behaviour as direct call) via something like reflection when `Shuffle`/`ShuffleUnsafe` are optimised.

Reply:
It will not.

Most of the other `*Native` or `*Estimate` operations are element wise, so they can be implemented "recursively" via the scalar calls.

You need some path that was recursive or otherwise guaranteed that this handling would be identical to what the JIT does. In many cases that could likely be:
```csharp
#if MONO
    if (Vector128.IsHardwareAccelerated)
    {
        return ShuffleNative();
    }
#endif

    return Shuffle();
```

However, for some of the pre-AVX fallbacks where the JIT can't trivially implement it; something else would need to be figured out.

Reply:
So does this mean that if we go from, say, SSE2-only R2R to JIT this function changes its behavior/return value in the same process? Isn't something we always avoid?